src/publicHolidayReminder.ts

```diff
@@ -41,33 +48,53 @@ const toDateStamp = date => {
 //       method: 'PUBLISH'
 //   },
 //
-const findRelevantEvents = (rawEvents, date) => {
-  return Object.values(rawEvents)
-    .filter(rawEvent => toDateStamp(rawEvent.start) === toDateStamp(date))
+const findRelevantEvents = (
+  calendarResponse: CalendarResponse,
+  date: Date
+): MyEvent[] => {
+  return Object.values(calendarResponse)
+    .filter(rawEvent => {
+      return (
+        isVEvent(rawEvent) && toDateStamp(rawEvent.start) === toDateStamp(date)
+      );
+    })
     .map(rawEvent => {
+      const vEvent = rawEvent as VEvent; // safe to cast here because we ran isVEvent in filter() above
```

Comment:
You can eliminate the cast by making the `filter` an explicit type guard:

```suggestion
    .filter((rawEvent): rawEvent is VEvent => {
      return (
        isVEvent(rawEvent) && toDateStamp(rawEvent.start) === toDateStamp(date)
      );
    })
    .map(rawEvent => {
      const vEvent = rawEvent;
```

Reply:
Ah ok. Somehow using a type guard function to do other things also passed me by. Thanks!

Reply:
`filter` is a classic gotcha, because by definition it always takes a typeguard but for safety reasons TypeScript is not allowed to infer typeguards because otherwise stuff could really get off the rails in subtle ways - a good rule of thumb is to always make the `filter` predicate function a type guard